beesdoo_website_shift/controllers/main.py

```diff
@@ -1,56 +1,437 @@
 # -*- coding: utf8 -*-
-from datetime import datetime
+
+# Copyright 2017-2018 Rémy Taymans <remytaymans@gmail.com>
+# Copyright 2017-2018 Thibault François
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from ast import literal_eval
+from copy import copy
+from datetime import datetime, timedelta
 from itertools import groupby
-from openerp import http
+
+from openerp import http, fields
 from openerp.http import request
+from openerp.tools import DEFAULT_SERVER_DATETIME_FORMAT as DATETIME_FORMAT
 
 from openerp.addons.beesdoo_shift.models.planning import float_to_time
 
-class ShiftPortalController(http.Controller):
+PERIOD = 28  # TODO: use the same constant as in 'beesdoo_shift'
```

Comment:
I think PERIOD should be in the module beesdoo_shift. But we can do the change later.